src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -182,7 +185,14 @@ module.exports = function<T, P, I, TI, PI, C, CX, PL>(
       }
     }
 
-    const priorityLevel = getPriorityContext();
+    // Check if the top-level element is an async wrapper component. If so, treat
+    // updates to the root as async. This is a bit weird but lets us avoid a separate
+    // `renderAsync` API.
+    const forceAsync = ReactFeatureFlags.enableAsyncSubtreeAPI &&
+      element != null &&
+      element.type != null &&
+      (element.type: any).unstable_asyncUpdates === true;
+    const priorityLevel = getPriorityContext(current, forceAsync);
```

Comment:
Instead of introducing a boolean flag that changes behavior, can you just override the priority level on the outside? Seems like a lot of cases need to care about `forceAsync` and it leaks. When you really could just do `if (forceAsync && priorityLevel < LowPriority) { priorityLevel = LowPriority; }` right here.